packages/blade/src/components/Checkbox/CheckboxGroup/useCheckboxGroup.ts

```diff
@@ -0,0 +1,89 @@
+/* eslint-disable @typescript-eslint/no-shadow */
+/* eslint-disable @typescript-eslint/prefer-nullish-coalescing */
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+
+import React from 'react';
+import { useControllableState } from '../../../hooks/useControllable';
+import { useId } from '../../../hooks/useId';
+import { useTheme } from '../../BladeProvider';
+import type { CheckboxGroupProps } from './CheckboxGroup';
+
+type UseCheckboxGroupProps = Pick<
+  CheckboxGroupProps,
+  | 'isDisabled'
+  | 'labelPosition'
+  | 'validationState'
+  | 'neccessityIndicator'
+  | 'name'
+  | 'value'
+  | 'defaultValue'
+  | 'onChange'
+>;
+
+const useCheckboxGroup = ({
+  value,
+  defaultValue,
+  isDisabled,
+  labelPosition,
+  onChange,
+  neccessityIndicator,
+  validationState,
+  name,
+}: UseCheckboxGroupProps) => {
```

Comment:
why no TS return type for function?

Reply:
IMO it's okay to let typescript infer it based on what I return. 
I don't find the value on doing the job what typescript already can. 

Q: Altho you might argue that "Now we don't have a static contract on what we should and should not return from this function". 

Ans: We have the contract, which is the `return {}` value itself. 


> Best TypeScript code is no TypeScript code.

```tsx

type UseCheckboxNativeReturn = {
  state: {
    isReactNative: boolean;
    isChecked: boolean;
    setChecked: React.Dispatch<React.SetStateAction<boolean>>;
  };
  inputProps: {
    onPress: (event: React.ChangeEvent<HTMLInputElement> | GestureResponderEvent) => void;
    name: string | undefined;
    value: string | undefined;
  };
};

type UseCheckboxWebReturn = {
  state: {
    isReactNative: boolean;
    isChecked: boolean;
    setChecked: React.Dispatch<React.SetStateAction<boolean>>;
  };
  ids: {
    errorTextId: string | undefined;
    helpTextId: string | undefined;
  };
  inputProps: {
    onChange: (event: React.ChangeEvent<HTMLInputElement> | GestureResponderEvent) => void;
    name: string | undefined;
    value: string | undefined;
    ref: React.RefObject<HTMLInputElement>;
    type: string;
    checked: boolean | undefined;
    disabled: boolean | undefined;
    required: boolean | undefined;
  };
};
```

Reply:
Depends on where and how the function is used. 

If it's a public function exposed to users and you want to make sure no one changes the return type randomly then it makes sense to explicitly add the return type.